src/components/OntologyBrowser/index.tsx

```diff
@@ -149,43 +197,57 @@ class OntologyModal extends React.Component<ModalProps, ModalState> {
   }
 
   render() {
+    const { isVisible } = this.props;
+    const { error, targetKeys, isLoading, treeSource } = this.state;
+    const dataSource = this.transfertDataSource;
+    const hasData = dataSource.length > 0;
+    const hasError = error != null;
     return (
       <Modal
         style={{ height: '80vh' }}
         title="HPO Onthology Browser"
-        visible={this.props.isVisible}
-        onOk={e => this.onApply(this.state.targetKeys)}
-        okText="Apply"
+        visible={isVisible}
+        onOk={() => this.onApply(targetKeys)}
+        okText={'Apply'}
+        okButtonProps={{ disabled: hasError }}
         onCancel={this.onCancel}
         cancelText="Cancel"
         width="90%"
       >
-        <Transfer
-          dataSource={this.transfertDataSource}
-          targetKeys={this.state.targetKeys}
-          onChange={this.onChange}
-          render={(item: TransferItem): RenderResult => item.title || item.key}
-          disabled={false}
-          showSelectAll={false}
-        >
-          {({ direction, onItemSelect, selectedKeys }) => {
-            if (direction === 'left') {
-              const checkedKeys = [...selectedKeys, ...this.state.targetKeys];
-              if (this.state.treeSource) {
+        {hasError ? (
+          displayError()
```

Comment:
I don't see value in this function jump, it doesn't make the code easier to understand

Reply:
Good point.